python/tk_desktop/rpc.py

```diff
@@ -8,29 +8,128 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
+"""
+This module exposes the components for two way communication between both
+processes of the Shotgun Desktop. There's the legacy communication protocol that works
+well enough for Python 2, MultiprocessingRPCServerThread and MultiprocessingRPCProxy,
+but is incompatible with Python 3. This is because the Python 3 version of
+multiprocessing.connection is forcing the pickle protocol to version 3, which does
+not exist in Python 2.
+
+To get around this, there's now a Http based implementation of the protocol that works
+between Python 2 and 3 processes. It is implemented by HttpRPCServerThread and
+HttpRPCProxy.
+
+The Http based implementation is a bit more complicated and involves a few
+more classes internally. The relationship between the classes are as follow:
+
+- HttpRPCServerThread owns the Listener.
+- The Listener is used to register methods and owns an HTTPServer that the Connection
+  will invoke.
+- The HttpRPCProxy owns a Connection object to send data to the HTTPServer.
+"""
 
 import os
 import sys
 import uuid
 import select
 import logging
 import threading
+import time
 import traceback
-import cPickle as pickle
 import multiprocessing.connection
 
+from sgtk.util import pickle
+from tank_vendor.six.moves.BaseHTTPServer import HTTPServer
+from tank_vendor.six.moves.SimpleHTTPServer import SimpleHTTPRequestHandler
+from tank_vendor.six.moves.urllib.request import urlopen
+from tank_vendor import six
+
 from sgtk import LogManager
 
-logger = LogManager.get_logger(__name__)
 
-# Only log debug messages if they are specifically requested.
-if "TK_DESKTOP_RPC_DEBUG" in os.environ:
-    logger.setLevel(logging.DEBUG)
-else:
-    logger.setLevel(logging.INFO)
+class Logger(object):
+    """
+    Wrapper around a logger object. It augments the logged information when
+    TK_DESKTOP_RPC_DEBUG is turned on.
```

Comment:
Why does this only handle `debug`, `error`, `exception` and `warning` and not `info` as well?

The comment here clearly states this is only used when debugging is enabled so I assume this is intentional. Scanning the code, it seem you are only logging debug messages within the rpc module. So I assume this logging object is only for use within the rpc module and does not affect normal logging coming from say an app running within a project process.

Reply:
I should have provided a bit more context. As far as the code review is concerned, TK_DESKTOP_RPC_DEBUG already existed before. I simply made it better.

I'll update the code.

Reply:
I forgot about `info`!